contracts/staking/feirari/AutoRewardsDistributorV2.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity ^0.8.0;
+
+import "./../ITribalChief.sol";
+import "../../refs/CoreRef.sol";
+import "../../external/Unitroller.sol";
+import "../StakingTokenWrapper.sol";
+import "./IRewardsDistributorAdmin.sol";
+
+/// @notice Controller Contract to set tribe per block in Rewards Distributor Admin on Rari
+contract AutoRewardsDistributorV2 is CoreRef {
+    /// @notice rewards distributor admin contract
+    IRewardsDistributorAdmin public rewardsDistributorAdmin;
+    /// @notice tribal chief contract
+    ITribalChief public immutable tribalChief;
+    /// @notice address of the underlying token for the cToken this contract controls rewards for
+    address public immutable underlying;
+    /// @notice boolean which decides the action to incentivize
+    bool public immutable isBorrowIncentivized;
+
+    /// @notice address of the comptroller, used to determine cToken
+    Unitroller public immutable comptroller;
+
+    /// @notice address of the stakedTokenWrapper
+    StakingTokenWrapper public immutable stakedTokenWrapper;
+
+    /// @notice address of the cToken this contract controls rewards for
+    address public cTokenAddress;
+
+    /// @notice reward index on tribal chief to grab this staked token wrapper's index
+    uint256 public tribalChiefRewardIndex;
+
+    event SpeedChanged(uint256 newSpeed);
+    event RewardsDistributorAdminChanged(IRewardsDistributorAdmin oldRewardsDistributorAdmin, IRewardsDistributorAdmin newRewardsDistributorAdmin);
+
+    /// @notice constructor function
+    /// @param coreAddress address of core contract
+    /// @param _rewardsDistributorAdmin address of rewards distributor admin contract
+    /// @param _tribalChief address of tribalchief contract
+    /// @param _stakedTokenWrapper the stakedTokenWrapper this contract controls rewards for
+    /// @param _underlying address of the underlying for the cToken
+    /// @param _isBorrowIncentivized boolean that incentivizes borrow or supply
+    /// @param _comptroller address of the comptroller contract
+    constructor(
+        address coreAddress,
+        IRewardsDistributorAdmin _rewardsDistributorAdmin,
+        ITribalChief _tribalChief,
+        StakingTokenWrapper _stakedTokenWrapper,
+        address _underlying,
+        bool _isBorrowIncentivized,
+        Unitroller _comptroller
+    ) CoreRef(coreAddress) {
+        isBorrowIncentivized = _isBorrowIncentivized;
+        underlying = _underlying;
+        stakedTokenWrapper = _stakedTokenWrapper;
+        rewardsDistributorAdmin = _rewardsDistributorAdmin;
+        tribalChief = _tribalChief;
+        comptroller = _comptroller;
+
+        _setContractAdminRole(keccak256("TRIBAL_CHIEF_ADMIN_ROLE"));
+    }
+
+    function init() external {
```

Comment:
This function can be called many times, is that bad?

Reply:
Good catch. This should be fixed, but it shouldn't have any adverse affect.

Reply:
Can't be called and pass bc the cToken check acts as an idempotency invariant